src/torchio/data/inference/aggregator.py

```diff
@@ -99,6 +103,26 @@ def _initialize_avgmask_tensor(self, batch: torch.Tensor) -> None:
             dtype=batch.dtype,
         )
 
+    @staticmethod
+    def _get_hann_window(patch_size):
+        hann_window_3d = torch.as_tensor([1])
+        # create a n-dim hann window
+        for spatial_dim, size in enumerate(patch_size):
+            window_shape = np.ones_like(patch_size)
+            window_shape[spatial_dim] = size
+            hann_window_1d = torch.hann_window(
+                size + 2,
+                periodic=False,
+            )
+            hann_window_1d = hann_window_1d[1:-1].view(*window_shape)
```

Comment:
Why do the first and last values need to be omitted?

Reply:
For a window of length >=2 (and this is assured given `torch.hann_window(size+2)`) the first and the last element of the result array will be 0 and this cause numerical problems (division by 0)

Reply:
I would suggest not discarding the zeros and adding an epsilon (maybe `torch.finfo(torch.float32).eps`) before dividing. What do you think?

Reply:
I had thought about this option but I've discarded it to avoid big numbers.
From a mathematical point of view, there shouldn't be any difference at all
Is there any reason why adding an epsilon is better than removing the first and last zero?
(It is worth noticing that this epsilon must be added to the N-dim hann window and not to the 1d, as this latter will cause epsilon^N to appear in the edges)